lib/string_reverse.rb

```diff
@@ -1,6 +1,46 @@
+require "pry"
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string passed into the string_reverse method.
+# Space complexity: Space complexity is O(1) because the amount of space occupied by the characters in the string plus each of the variables does not change.
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  i_start = 0
+  i_end = -1
+  a = ""

```

Comment:
Two thoughts RE variable names:
1) Is the `i_` prefix on `i_start` and `i_end` redundant? Perhaps `start` and `end` would be more concise. 🙂
2) you might replace `a`, `z`, and `t` with more descriptive variable names, as this will likely make your code easier [for others] to understand.